src/chainstate/stacks/db/blocks.rs

```diff
@@ -4282,6 +4282,251 @@ impl StacksChainState {
         Ok(parent_miner)
     }
 
+    /// Called in both follower and miner block assembly paths.
+    /// Returns clarity_tx, list of receipts, microblock execution cost,
+    /// microblock fees, microblock burns, list of microblock tx receipts,
+    /// miner rewards tuples, the stacks epoch id, and a boolean that
+    /// represents whether the epoch transition has been applied.
+    pub fn setup_block<'a>(
+        chainstate_tx: &'a mut ChainstateTx,
+        clarity_instance: &'a mut ClarityInstance,
+        burn_dbconn: &'a dyn BurnStateDB,
+        conn: &Connection,
+        chain_tip: &StacksHeaderInfo,
+        burn_tip: BurnchainHeaderHash,
+        burn_tip_height: u32,
+        parent_consensus_hash: ConsensusHash,
+        parent_header_hash: BlockHeaderHash,
+        parent_microblocks: &Vec<StacksMicroblock>,
+        mainnet: bool,
+        miner_id_opt: Option<usize>,
+    ) -> Result<
+        (
```

Comment:
Do you think we could maybe consolidate the returned tuple into a `struct`, so it's clear to the caller what each field is?